Fail to build on Windows 64 bit

I currently have Python 2.7 installed and just purchased a GTX 980 (after having a 7970) to be able to use cudamat to speed up my theano neural nets.

When I run "python setup.py install" I get this error:

> $ python setup.py install
> running install
> running bdist_egg
> running egg_info
> writing cudamat.egg-info\PKG-INFO
> writing top-level names to cudamat.egg-info\top_level.txt
> writing dependency_links to cudamat.egg-info\dependency_links.txt
> reading manifest file 'cudamat.egg-info\SOURCES.txt'
> writing manifest file 'cudamat.egg-info\SOURCES.txt'
> installing library code to build\bdist.win-amd64\egg
> running install_lib
> running build_py
> creating build\lib.win-amd64-2.7
> creating build\lib.win-amd64-2.7\cudamat
> copying cudamat\cudamat.py -> build\lib.win-amd64-2.7\cudamat
> copying cudamat\learn.py -> build\lib.win-amd64-2.7\cudamat
> copying cudamat\__init__.py -> build\lib.win-amd64-2.7\cudamat
> copying cudamat\cudamat.cu -> build\lib.win-amd64-2.7\cudamat
> copying cudamat\cudamat_kernels.cu -> build\lib.win-amd64-2.7\cudamat
> copying cudamat\learn.cu -> build\lib.win-amd64-2.7\cudamat
> copying cudamat\learn_kernels.cu -> build\lib.win-amd64-2.7\cudamat
> copying cudamat\rnd_multipliers_32bit.txt -> build\lib.win-amd64-2.7\cudamat
> running build_ext
> building 'cudamat.libcudamat' extension
> error: Don't know how to compile cudamat/cudamat.cu to build\temp.win-amd64-> 2.7\Release\cudamat/cudamat.obj

Alternatively, running "pip install" gives me:

> $ pip install
> ←[33mYou must give at least one requirement to install (see "pip help install")←[0m

What is going wrong?

Alright, seems that the nvcc is finally working, although would love to know quick command that indicates if it does as intended. When i try to use the 
```
nvcc -O --ptxas-options=-v -o libcudamat.dll --shared cudamat.cu cudamat_kernels.cu -lcublas
nvcc -O --ptxas-options=-v -o libcudalearn.dll --shared learn.cu learn_kernels.cu -lcublas
```
commands, i get the error: 
```
c1xx : fatal error C1083: Cannot open source file: 'cudamat.cu': No such file or directory
c1xx : fatal error C1083: Cannot open source file: 'learn.cu': No such file or directory
```
So the following question is, which directory is the one i should launch it from ?

And here's the result:
```c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\host_config.h(181) : fatal error C1083: Cannot open include file: 'crtdefs.h': No such file or directory```

So it did work, by including that line ```-I"C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include"```
The results are:
```cudamat.cu
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions_decls.h : warning C4819: The file conta
ins a character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent d
ata loss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
ptxas info    : 0 bytes gmem
cudamat_kernels.cu
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions_decls.h : warning C4819: The file conta
ins a character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent d
ata loss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

ptxas info    : 0 bytes gmem
ptxas info    : Compiling entry function '_Z10kApplyTanhPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z10kApplyTanhPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 52 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z16kMultByRowVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z16kMultByRowVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z7kEqualsPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z7kEqualsPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z5kSqrtPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z5kSqrtPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 52 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z15kRandomGaussianPjPyPfj' for 'sm_20'
ptxas info    : Function properties for _Z15kRandomGaussianPjPyPfj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 14 registers, 60 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z13kAssignScalarPffj' for 'sm_20'
ptxas info    : Function properties for _Z13kAssignScalarPffj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 7 registers, 48 bytes cmem[0]
ptxas info    : Compiling entry function '_Z14kMaxColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z14kMaxColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 128 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z4kLogPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z4kLogPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 7 registers, 52 bytes cmem[0]
ptxas info    : Compiling entry function '_Z9kSubtractPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z9kSubtractPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z8kMaximumPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z8kMaximumPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z13kAddColVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z13kAddColVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z9kLessThanPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z9kLessThanPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z16kSetSelectedRowsPfS_S_iii' for 'sm_20'
ptxas info    : Function properties for _Z16kSetSelectedRowsPfS_S_iii
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 128 bytes smem, 68 bytes cmem[0], 16 bytes cmem[16]
ptxas info    : Compiling entry function '_Z17kArgMaxColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z17kArgMaxColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 256 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z5kSignPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z5kSignPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 52 bytes cmem[0]
ptxas info    : Compiling entry function '_Z11kAddColMultPfS_S_fjj' for 'sm_20'
ptxas info    : Function properties for _Z11kAddColMultPfS_S_fjj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 68 bytes cmem[0]
ptxas info    : Compiling entry function '_Z12kGreaterThanPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z12kGreaterThanPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z6kGammaPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z6kGammaPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 17 registers, 52 bytes cmem[0], 64 bytes cmem[16]
ptxas info    : Compiling entry function '_Z11kSeedRandomPjPyj' for 'sm_20'
ptxas info    : Function properties for _Z11kSeedRandomPjPyj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 8 registers, 52 bytes cmem[0]
ptxas info    : Compiling entry function '_Z5kMultPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z5kMultPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z14kMinColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z14kMinColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 128 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z9kApplyAbsPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z9kApplyAbsPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 52 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z15kDivByRowVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z15kDivByRowVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 17 registers, 64 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z8kMinimumPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z8kMinimumPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z10kPowMatrixPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z10kPowMatrixPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 60 bytes cmem[0], 40 bytes cmem[16]
ptxas info    : Compiling entry function '_Z12kSetRowSlicePfS_iiii' for 'sm_20'
ptxas info    : Function properties for _Z12kSetRowSlicePfS_iiii
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z10kAddScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z10kAddScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z17kArgMinColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z17kArgMinColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 256 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z19kApplySoftThresholdPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z19kApplySoftThresholdPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z15kDivByColVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z15kDivByColVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 17 registers, 64 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z13kEqualsScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z13kEqualsScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z4kPowPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z4kPowPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 18 registers, 60 bytes cmem[0], 36 bytes cmem[16]
ptxas info    : Compiling entry function '_Z12kGetRowSlicePfS_iiii' for 'sm_20'
ptxas info    : Function properties for _Z12kGetRowSlicePfS_iiii
    0 bytes stack frame, 0 bytes spill stores, 0 C:\Users\Don\Documents\GitHub\cudamat\cudamat^C [win-support]> nvcc -O
--ptxas-options=-v -o libcudamat.dll --shared cudamat.cu cudamat_kernels.cu -lcublas -I"C:\Program Files (x86)\Microsoft
 Visual Studio 12.0\VC\include"
cudamat.cu
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions_decls.h : warning C4819: The file conta
ins a character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent d
ata loss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
ptxas info    : 0 bytes gmem
cudamat_kernels.cu
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions_decls.h : warning C4819: The file conta
ins a character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent d
ata loss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
c:\program files\nvidia gpu computing toolkit\cuda\v7.0\include\device_functions.h : warning C4819: The file contains a
character that cannot be represented in the current code page (1257). Save the file in Unicode format to prevent data lo
ss
cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(747): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

cudamat_kernels.cu(771): warning: division by zero

ptxas info    : 0 bytes gmem
ptxas info    : Compiling entry function '_Z10kApplyTanhPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z10kApplyTanhPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 52 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z16kMultByRowVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z16kMultByRowVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z7kEqualsPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z7kEqualsPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z5kSqrtPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z5kSqrtPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 52 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z15kRandomGaussianPjPyPfj' for 'sm_20'
ptxas info    : Function properties for _Z15kRandomGaussianPjPyPfj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 14 registers, 60 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z13kAssignScalarPffj' for 'sm_20'
ptxas info    : Function properties for _Z13kAssignScalarPffj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 7 registers, 48 bytes cmem[0]
ptxas info    : Compiling entry function '_Z14kMaxColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z14kMaxColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 128 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z4kLogPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z4kLogPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 7 registers, 52 bytes cmem[0]
ptxas info    : Compiling entry function '_Z9kSubtractPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z9kSubtractPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z8kMaximumPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z8kMaximumPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z13kAddColVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z13kAddColVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z9kLessThanPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z9kLessThanPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z16kSetSelectedRowsPfS_S_iii' for 'sm_20'
ptxas info    : Function properties for _Z16kSetSelectedRowsPfS_S_iii
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 128 bytes smem, 68 bytes cmem[0], 16 bytes cmem[16]
ptxas info    : Compiling entry function '_Z17kArgMaxColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z17kArgMaxColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 256 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z5kSignPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z5kSignPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 52 bytes cmem[0]
ptxas info    : Compiling entry function '_Z11kAddColMultPfS_S_fjj' for 'sm_20'
ptxas info    : Function properties for _Z11kAddColMultPfS_S_fjj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 68 bytes cmem[0]
ptxas info    : Compiling entry function '_Z12kGreaterThanPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z12kGreaterThanPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z6kGammaPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z6kGammaPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 17 registers, 52 bytes cmem[0], 64 bytes cmem[16]
ptxas info    : Compiling entry function '_Z11kSeedRandomPjPyj' for 'sm_20'
ptxas info    : Function properties for _Z11kSeedRandomPjPyj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 8 registers, 52 bytes cmem[0]
ptxas info    : Compiling entry function '_Z5kMultPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z5kMultPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z14kMinColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z14kMinColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 128 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z9kApplyAbsPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z9kApplyAbsPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 52 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z15kDivByRowVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z15kDivByRowVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 17 registers, 64 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z8kMinimumPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z8kMinimumPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z10kPowMatrixPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z10kPowMatrixPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 60 bytes cmem[0], 40 bytes cmem[16]
ptxas info    : Compiling entry function '_Z12kSetRowSlicePfS_iiii' for 'sm_20'
ptxas info    : Function properties for _Z12kSetRowSlicePfS_iiii
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z10kAddScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z10kAddScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z17kArgMinColumnwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z17kArgMinColumnwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 256 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z19kApplySoftThresholdPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z19kApplySoftThresholdPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z15kDivByColVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z15kDivByColVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 17 registers, 64 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z13kEqualsScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z13kEqualsScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z4kPowPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z4kPowPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 18 registers, 60 bytes cmem[0], 36 bytes cmem[16]
ptxas info    : Compiling entry function '_Z12kGetRowSlicePfS_iiii' for 'sm_20'
ptxas info    : Function properties for _Z12kGetRowSlicePfS_iiii
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z13kDivideScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z13kDivideScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 60 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z11kMaxRowwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z11kMaxRowwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 128 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z4kExpPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z4kExpPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 7 registers, 52 bytes cmem[0]
ptxas info    : Compiling entry function '_Z7kDividePfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z7kDividePfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 60 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z14kMaximumScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z14kMaximumScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z13kAddRowVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z13kAddRowVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z15kLessThanScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z15kLessThanScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z6kWherePfS_S_S_j' for 'sm_20'
ptxas info    : Function properties for _Z6kWherePfS_S_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 68 bytes cmem[0]
ptxas info    : Compiling entry function '_Z14kArgMaxRowwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z14kArgMaxRowwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 256 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z13kApplySigmoidPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z13kApplySigmoidPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 14 registers, 52 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z16kMultByColVectorPfS_S_jj' for 'sm_20'
ptxas info    : Function properties for _Z16kMultByColVectorPfS_S_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 13 registers, 64 bytes cmem[0]
ptxas info    : Compiling entry function '_Z18kGreaterThanScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z18kGreaterThanScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z9kLogGammaPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z9kLogGammaPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 15 registers, 52 bytes cmem[0], 196 bytes cmem[16]
ptxas info    : Compiling entry function '_Z14kRandomUniformPjPyPfj' for 'sm_20'
ptxas info    : Function properties for _Z14kRandomUniformPjPyPfj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 22 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z11kMultScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z11kMultScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z11kMinRowwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z11kMinRowwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 16 registers, 128 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z17kApplyLog1PlusExpPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z17kApplyLog1PlusExpPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 8 registers, 52 bytes cmem[0], 4 bytes cmem[16]
ptxas info    : Compiling entry function '_Z4kAddPfS_S_j' for 'sm_20'
ptxas info    : Function properties for _Z4kAddPfS_S_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 11 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z14kMinimumScalarPffS_j' for 'sm_20'
ptxas info    : Function properties for _Z14kMinimumScalarPffS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 9 registers, 60 bytes cmem[0]
ptxas info    : Compiling entry function '_Z11kReciprocalPfS_j' for 'sm_20'
ptxas info    : Function properties for _Z11kReciprocalPfS_j
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 14 registers, 52 bytes cmem[0], 8 bytes cmem[16]
ptxas info    : Compiling entry function '_Z10kTransposePfS_ii' for 'sm_20'
ptxas info    : Function properties for _Z10kTransposePfS_ii
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 10 registers, 1088 bytes smem, 56 bytes cmem[0]
ptxas info    : Compiling entry function '_Z11kSelectRowsPfS_S_iii' for 'sm_20'
ptxas info    : Function properties for _Z11kSelectRowsPfS_S_iii
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 128 bytes smem, 68 bytes cmem[0], 16 bytes cmem[16]
ptxas info    : Compiling entry function '_Z14kArgMinRowwisePfS_jj' for 'sm_20'
ptxas info    : Function properties for _Z14kArgMinRowwisePfS_jj
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 20 registers, 256 bytes smem, 56 bytes cmem[0], 4 bytes cmem[16]
LINK : fatal error LNK1104: cannot open file 'LIBCMT.lib'
```
Googling about LIBCMT now

Okay, did that and here's what came out: 

    F:\github\cudamat>python setup.py build_ext
    running build_ext
    DEBUG BEXT <distutils.msvc9compiler.MSVCCompiler instance at 0x0000000003333408>
    building 'cudamat.libcudamat' extension
    DEBUG SPAWN <__main__.CUDA_build_ext instance at 0x00000000033330C8> ['C:\\Users\\Hendrik\\AppData\\Local\\Programs\\Common\\Microso
    ft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\cl.exe', '/c', '/nologo', '/Ox', '/MD', '/W3', '/GS-', '/DNDEBUG', '-IF:\\Anaconda\\
    include', '-IF:\\Anaconda\\PC', '/Tccudamat/cudamat.cu', '/Fobuild\\temp.win-amd64-2.7\\Release\\cudamat/cudamat.obj', '-O', '--ptxa
    s-options=-v', '--compiler-options', "'-fPIC'"]
    C:\Users\Hendrik\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\cl.exe /c /nologo /Ox /MD /W3 /GS- /
    DNDEBUG -IF:\Anaconda\include -IF:\Anaconda\PC /Tccudamat/cudamat.cu /Fobuild\temp.win-amd64-2.7\Release\cudamat/cudamat.obj -O --pt
    xas-options=-v --compiler-options '-fPIC'
    cl : Command line warning D9035 : option 'O' has been deprecated and will be removed in a future release
    cl : Command line warning D9002 : ignoring unknown option '--ptxas-options=-v'
    cl : Command line warning D9002 : ignoring unknown option '--compiler-options'
    cl : Command line warning D9024 : unrecognized source file type ''-fPIC'', object file assumed
    cl : Command line warning D9027 : source file ''-fPIC'' ignored
    cudamat.cu
    cudamat/cudamat.cu(3) : fatal error C1083: Cannot open include file: 'cublas.h': No such file or directory
    error: command 'C:\\Users\\Hendrik\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\cl.exe'
Apparently it's trying to launch cl.exe instead of nvcc.exe to compile the .cu file. I suppose that wouldn't work even if the compiler was able to find the `cublas.h` header.

When I run the commands, this is what happens:

    F:\github\cudamat\cudamat>nvcc --compiler-options=/c,/nologo,/Ox,/MD,/W3,/GS-,/DNDEBUG,-IF:\Anaconda\include,-IF:\Anaconda\PC cudamat.cu -o cudamat/cudamat.obj -O --ptxas-options=-v
    cudamat.cu
    ptxas info    : 0 bytes gmem
    cudamat.cu(179) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(200) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(275) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(316) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(322) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cl : Befehlszeile warning D9027 : Quelldatei "F:/2014HE~1/Temp/tmpxft_00004248_00000000-32_cudamat_dlink.obj" wird ignoriert.
    cl : Befehlszeile warning D9027 : Quelldatei "F:/2014HE~1/Temp/tmpxft_00004248_00000000-28_cudamat.obj" wird ignoriert.
    cl : Befehlszeile warning D9021 : Keine Aktion durchgeführt
    
    
    F:\github\cudamat\cudamat>nvcc -c --compiler-options=/nologo,/Ox,/MD,/W3,/GS-,/DNDEBUG,-IF:\Anaconda\include,-IF:\Anaconda\PC cudamat.cu -o cudamat/cudamat.obj -O --ptxas-options=-v
    cudamat.cu
    ptxas info    : 0 bytes gmem
    cudamat.cu(179) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(200) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(275) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(316) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(322) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    F:\2014HE~1\Temp\tmpxft_00006e24_00000000-23_cudamat.ii : fatal error C1083: Datei (vom Compiler generiert) kann nicht geöffnet werden: "cudamat/cudamat.obj": No such file or directory

I've changed the second command to comply with the folder structur and it has successfully created the .obj file.

    F:\github\cudamat\cudamat>nvcc -c --compiler-options=/nologo,/Ox,/MD,/W3,/GS-,/DNDEBUG,-IF:\Anaconda\include,-IF:\Anaconda\PC cudamat.cu -o cudamat.obj -O --ptxas-options=-v
    cudamat.cu
    ptxas info    : 0 bytes gmem
    cudamat.cu(179) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(200) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(275) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(316) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat.cu(322) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'

Okay, `nvcc` now complains about not being able to find a suitable MSVS installation. Executing the command manually however works like a charm. 

    F:\github\cudamat>python setup.py build_ext
    running build_ext
    DEBUG BEXT <distutils.msvc9compiler.MSVCCompiler instance at 0x000000000334EE08>
    building 'cudamat.libcudamat' extension
    DEBUG SPAWN <__main__.CUDA_build_ext instance at 0x000000000334E6C8> ['C:\\Users\\Hendrik\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\cl.exe', '/c', '/nol
    ogo', '/Ox', '/MD', '/W3', '/GS-', '/DNDEBUG', '-IF:\\Anaconda\\include', '-IF:\\Anaconda\\PC', '/Tccudamat/cudamat.cu', '/Fobuild\\temp.win-amd64-2.7\\Release\\cudamat/cudamat.obj', '-O', '--ptxas-op
    tions=-v', '--compiler-options', "'-fPIC'"]
    C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\bin\nvcc.exe -c -IF:\Anaconda\include -IF:\Anaconda\PC cudamat/cudamat.cu -o build\temp.win-amd64-2.7\Release\cudamat/cudamat.obj -O --ptxas-opt
    ions=-v --compiler-options=/nologo,/Ox,/MD,/W3,/GS-,/DNDEBUG
    nvcc fatal   : nvcc cannot find a supported version of Microsoft Visual Studio. Only the versions 2010, 2012, and 2013 are supported
    error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v7.0\\bin\\nvcc.exe' failed with exit status 1
    
    F:\github\cudamat>python setup.py build_ext
    
    F:\github\cudamat>nvcc -c -IF:\Anaconda\include -IF:\Anaconda\PC cudamat/cudamat.cu -o build\temp.win-amd64-2.7\Release\cudamat/cudamat.obj -O --ptxas-options=-v --compiler-options=/nologo,/Ox,/MD,/W3
    ,/GS-,/DNDEBUG
    cudamat.cu
    ptxas info    : 0 bytes gmem
    cudamat/cudamat.cu(179) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat/cudamat.cu(200) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat/cudamat.cu(275) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat/cudamat.cu(316) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'
    cudamat/cudamat.cu(322) : warning C4018: '>': Konflikt zwischen 'signed' und 'unsigned'

That did the trick... strange. 

What I have now:

        elif self.compiler.compiler_type == 'msvc':
            print "DEBUG command before: %s" % " ".join(cmd)
            from distutils.spawn import find_executable
            cmd[:1] = ['nvcc', '--compiler-bindir',
                       os.path.dirname(find_executable("cl.exe", PATH)) or cmd[0]]
            del cmd[-2:]
            for idx, c in enumerate(cmd):
                if c == '/c': cmd[idx] = '-c'
                if c == '/DLL': cmd[idx] = '--shared'
                if c.startswith('/Tc'): cmd[idx] = c[3:]
                if c.startswith('/Fo'): cmd[idx:idx+1] = ['-o', c[3:]]
                if c.startswith('/LIBPATH:'): cmd[idx] = '-L' + c[9:]
                if c.startswith('/OUT:'): cmd[idx:idx+1] = ['-o', c[5:]]
                if c == 'cublas.lib': cmd[idx] = '-lcublas'
            cmd = [c for c in cmd if c[0] != '/'] + ['--compiler-options=' + ','.join(c for c in cmd if c[0] == '/')]        
            print "DEBUG command after: %s" % " ".join(cmd)


Still there are the linker issues. Here's the latest output (stripped from the output of the successful compiler steps): 

    ...
    DEBUG SPAWN <__main__.CUDA_build_ext instance at 0x00000000032F3388> ['C:\\Users\\Hendrik\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\link.exe', '/DLL', '
    /nologo', '/INCREMENTAL:NO', '/LIBPATH:F:\\Anaconda\\libs', '/LIBPATH:F:\\Anaconda\\PCbuild\\amd64', 'cublas.lib', '/EXPORT:initlibcudamat', 'build\\temp.win-amd64-2.7\\Release\\cudamat/cudamat.obj',
    'build\\temp.win-amd64-2.7\\Release\\cudamat/cudamat_kernels.obj', '/OUT:build\\lib.win-amd64-2.7\\cudamat\\libcudamat.pyd', '/IMPLIB:build\\temp.win-amd64-2.7\\Release\\cudamat\\libcudamat.lib', '/MA
    NIFESTFILE:build\\temp.win-amd64-2.7\\Release\\cudamat\\libcudamat.pyd.manifest']
    DEBUG command before: C:\Users\Hendrik\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:F:\Anaconda\libs /LIBPATH:F:\Anacon
    da\PCbuild\amd64 cublas.lib /EXPORT:initlibcudamat build\temp.win-amd64-2.7\Release\cudamat/cudamat.obj build\temp.win-amd64-2.7\Release\cudamat/cudamat_kernels.obj /OUT:build\lib.win-amd64-2.7\cudama
    t\libcudamat.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\cudamat\libcudamat.lib /MANIFESTFILE:build\temp.win-amd64-2.7\Release\cudamat\libcudamat.pyd.manifest
    DEBUG command after: nvcc --compiler-bindir C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64 --shared -LF:\Anaconda\libs -LF:\Anaconda\PCbuild\amd64 -lcublas build\temp.win-amd64-2.7\R
    elease\cudamat/cudamat.obj build\temp.win-amd64-2.7\Release\cudamat/cudamat_kernels.obj -o build\lib.win-amd64-2.7\cudamat\libcudamat.pyd --compiler-options=/nologo,/INCREMENTAL:NO,/EXPORT:initlibcuda
    mat
    C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\bin\nvcc.exe --compiler-bindir "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64" --shared -LF:\Anaconda\libs -LF:\Anaconda\PCbu
    ild\amd64 -lcublas build\temp.win-amd64-2.7\Release\cudamat/cudamat.obj build\temp.win-amd64-2.7\Release\cudamat/cudamat_kernels.obj -o build\lib.win-amd64-2.7\cudamat\libcudamat.pyd --compiler-option
    s=/nologo,/INCREMENTAL:NO,/EXPORT:initlibcudamat
    cudamat.obj
    cudamat_kernels.obj
    cl : Befehlszeile warning D9002 : Unbekannte Option "/EXPORT:initlibcudamat" wird ignoriert.
    cl : Befehlszeile warning D9002 : Unbekannte Option "/EXPORT:initlibcudamat" wird ignoriert.
    MSVCRT.lib(MSVCR120.dll) : error LNK2005: malloc ist bereits in LIBCMT.lib(malloc.obj) definiert.
    MSVCRT.lib(MSVCR120.dll) : error LNK2005: fclose ist bereits in LIBCMT.lib(fclose.obj) definiert.
    MSVCRT.lib(MSVCR120.dll) : error LNK2005: printf ist bereits in LIBCMT.lib(printf.obj) definiert.
       Bibliothek "build/lib.win-amd64-2.7/cudamat/libcudamat.lib" und Objekt "build/lib.win-amd64-2.7/cudamat/libcudamat.exp" werden erstellt.
    LINK : warning LNK4098: Standardbibliothek "MSVCRT" steht in Konflikt mit anderen Bibliotheken; /NODEFAULTLIB:Bibliothek verwenden.
    build/lib.win-amd64-2.7/cudamat/libcudamat.pyd : fatal error LNK1169: Mindestens ein mehrfach definiertes Symbol gefunden.
    error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v7.0\\bin\\nvcc.exe' failed with exit status 2

Okay... here's the new output (again, stripped from the successful compiler output): 

    ...
    DEBUG SPAWN <__main__.CUDA_build_ext instance at 0x0000000003396388> ['C:\\Users\\Hendrik\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\link.exe', '/DLL', '
    /nologo', '/INCREMENTAL:NO', '/LIBPATH:F:\\Anaconda\\libs', '/LIBPATH:F:\\Anaconda\\PCbuild\\amd64', 'cublas.lib', '/EXPORT:initlibcudamat', 'build\\temp.win-amd64-2.7\\Release\\cudamat/cudamat.obj',
    'build\\temp.win-amd64-2.7\\Release\\cudamat/cudamat_kernels.obj', '/OUT:build\\lib.win-amd64-2.7\\cudamat\\libcudamat.pyd', '/IMPLIB:build\\temp.win-amd64-2.7\\Release\\cudamat\\libcudamat.lib', '/MA
    NIFESTFILE:build\\temp.win-amd64-2.7\\Release\\cudamat\\libcudamat.pyd.manifest']
    DEBUG command before: C:\Users\Hendrik\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:F:\Anaconda\libs /LIBPATH:F:\Anacon
    da\PCbuild\amd64 cublas.lib /EXPORT:initlibcudamat build\temp.win-amd64-2.7\Release\cudamat/cudamat.obj build\temp.win-amd64-2.7\Release\cudamat/cudamat_kernels.obj /OUT:build\lib.win-amd64-2.7\cudama
    t\libcudamat.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\cudamat\libcudamat.lib /MANIFESTFILE:build\temp.win-amd64-2.7\Release\cudamat\libcudamat.pyd.manifest
    DEBUG command after: nvcc --compiler-bindir C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64 --shared -LF:\Anaconda\libs -LF:\Anaconda\PCbuild\amd64 -lcublas build\temp.win-amd64-2.7\R
    elease\cudamat/cudamat.obj build\temp.win-amd64-2.7\Release\cudamat/cudamat_kernels.obj -o build\lib.win-amd64-2.7\cudamat\libcudamat.pyd --linker-options=/nologo,/INCREMENTAL:NO,/EXPORT:initlibcudama
    t
    C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\bin\nvcc.exe --compiler-bindir "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64" --shared -LF:\Anaconda\libs -LF:\Anaconda\PCbu
    ild\amd64 -lcublas build\temp.win-amd64-2.7\Release\cudamat/cudamat.obj build\temp.win-amd64-2.7\Release\cudamat/cudamat_kernels.obj -o build\lib.win-amd64-2.7\cudamat\libcudamat.pyd --linker-options=
    /nologo,/INCREMENTAL:NO,/EXPORT:initlibcudamat
    cudamat.obj
    cudamat_kernels.obj
    MSVCRT.lib(MSVCR120.dll) : error LNK2005: malloc ist bereits in LIBCMT.lib(malloc.obj) definiert.
    MSVCRT.lib(MSVCR120.dll) : error LNK2005: fclose ist bereits in LIBCMT.lib(fclose.obj) definiert.
    MSVCRT.lib(MSVCR120.dll) : error LNK2005: printf ist bereits in LIBCMT.lib(printf.obj) definiert.
    LINK : error LNK2001: Nicht aufgelöstes externes Symbol "initlibcudamat".
    build/lib.win-amd64-2.7/cudamat/libcudamat.lib : fatal error LNK1120: 1 nicht aufgelöste Externe
    error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v7.0\\bin\\nvcc.exe' failed with exit status 2

Fights tooth and nail... 

    ...
    DEBUG SPAWN <__main__.CUDA_build_ext instance at 0x000000000326D388> ['C:\\Users\\Hendrik\\AppData\\Local\\Programs\\Common\\Microsoft\\Visual C++ for Python\\9.0\\VC\\Bin\\amd64\\link.exe', '/DLL', '
    /nologo', '/INCREMENTAL:NO', '/LIBPATH:F:\\Anaconda\\libs', '/LIBPATH:F:\\Anaconda\\PCbuild\\amd64', 'cublas.lib', '/EXPORT:initlibcudalearn', 'build\\temp.win-amd64-2.7\\Release\\cudamat/learn.obj',
    'build\\temp.win-amd64-2.7\\Release\\cudamat/learn_kernels.obj', '/OUT:build\\lib.win-amd64-2.7\\cudamat\\libcudalearn.pyd', '/IMPLIB:build\\temp.win-amd64-2.7\\Release\\cudamat\\libcudalearn.lib', '/
    MANIFESTFILE:build\\temp.win-amd64-2.7\\Release\\cudamat\\libcudalearn.pyd.manifest']
    DEBUG command before: C:\Users\Hendrik\AppData\Local\Programs\Common\Microsoft\Visual C++ for Python\9.0\VC\Bin\amd64\link.exe /DLL /nologo /INCREMENTAL:NO /LIBPATH:F:\Anaconda\libs /LIBPATH:F:\Anacon
    da\PCbuild\amd64 cublas.lib /EXPORT:initlibcudalearn build\temp.win-amd64-2.7\Release\cudamat/learn.obj build\temp.win-amd64-2.7\Release\cudamat/learn_kernels.obj /OUT:build\lib.win-amd64-2.7\cudamat\
    libcudalearn.pyd /IMPLIB:build\temp.win-amd64-2.7\Release\cudamat\libcudalearn.lib /MANIFESTFILE:build\temp.win-amd64-2.7\Release\cudamat\libcudalearn.pyd.manifest
    DEBUG command after: nvcc --compiler-bindir C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64 --shared -LF:\Anaconda\libs -LF:\Anaconda\PCbuild\amd64 -lcublas build\temp.win-amd64-2.7\R
    elease\cudamat/learn.obj build\temp.win-amd64-2.7\Release\cudamat/learn_kernels.obj -o build\lib.win-amd64-2.7\cudamat\libcudalearn.pyd --linker-options=/nologo,/INCREMENTAL:NO
    C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\bin\nvcc.exe --compiler-bindir "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64" --shared -LF:\Anaconda\libs -LF:\Anaconda\PCbu
    ild\amd64 -lcublas build\temp.win-amd64-2.7\Release\cudamat/learn.obj build\temp.win-amd64-2.7\Release\cudamat/learn_kernels.obj -o build\lib.win-amd64-2.7\cudamat\libcudalearn.pyd --linker-options=/n
    ologo,/INCREMENTAL:NO
    learn.obj
    learn_kernels.obj
    MSVCRT.lib(MSVCR120.dll) : error LNK2005: printf ist bereits in LIBCMT.lib(printf.obj) definiert.
       Bibliothek "build/lib.win-amd64-2.7/cudamat/libcudalearn.lib" und Objekt "build/lib.win-amd64-2.7/cudamat/libcudalearn.exp" werden erstellt.
    LINK : warning LNK4098: Standardbibliothek "MSVCRT" steht in Konflikt mit anderen Bibliotheken; /NODEFAULTLIB:Bibliothek verwenden.
    build/lib.win-amd64-2.7/cudamat/libcudalearn.pyd : fatal error LNK1169: Mindestens ein mehrfach definiertes Symbol gefunden.
    error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v7.0\\bin\\nvcc.exe' failed with exit status 2

As you command ;-) 

    F:\github\cudamat\test>python test_cudamat.py
    ..................................F.....................
    ======================================================================
    FAIL: __main__.test_pow_matrix
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "C:\Users\Hendrik\AppData\Roaming\Python\Python27\site-packages\nose-1.3.4-py2.7.egg\nose\case.py", line 197, in runTest
        self.test(*self.arg)
      File "test_cudamat.py", line 791, in test_pow_matrix
        assert np.max(np.abs(c - m1.numpy_array)) < 10**-2, "Error in cudamat.pow exceeded threshold"
    AssertionError: Error in cudamat.pow exceeded threshold
    
    ----------------------------------------------------------------------
    Ran 56 tests in 2.505s
    
    FAILED (failures=1)
    
    F:\github\cudamat\test>python test_learn.py
    .
    ----------------------------------------------------------------------
    Ran 1 test in 0.179s
    
    OK

Looks like it's working. I guess, the failure is caused by something else.

--- a/cudamat/cudamat.cu
+++ b/cudamat/cudamat.cu
@@ -22,13 +22,13 @@ inline bool checkCUDAError() {
     return cudaSuccess != err;
 }
 
-extern const char* get_last_cuda_error() {
+EXPORT const char* get_last_cuda_error() {
     cudaError_t err = cudaGetLastError();
 
     return cudaGetErrorString( err);
 }
 
-extern int cublas_init() {
+EXPORT int cublas_init() {
     cublasInit();
     if (check_cublas_error())
         return CUBLAS_ERROR;
@@ -36,7 +36,7 @@ extern int cublas_init() {
         return 0;
 }
 
-extern int cublas_shutdown() {
+EXPORT int cublas_shutdown() {
     cublasShutdown();
     cudaThreadExit();
 
@@ -44,7 +44,7 @@ extern int cublas_shutdown() {
 }
 
 
-extern int cuda_set_device(int deviceId) {
+EXPORT int cuda_set_device(int deviceId) {
     cudaSetDevice(deviceId);
     
     if (checkCUDAError())
@@ -53,7 +53,7 @@ extern int cuda_set_device(int deviceId) {
         return 0;
 }
 
-extern int init_random(rnd_struct* rnd_state, int seed, char* cudamatpath) {
+EXPORT int init_random(rnd_struct* rnd_state, int seed, char* cudamatpath) {
     unsigned int * host_mults;
     host_mults = (unsigned int*)malloc(NUM_RND_STREAMS * sizeof(unsigned int));
     FILE * pFile;
@@ -85,15 +85,15 @@ extern int init_random(rnd_struct* rnd_state, int seed, char* cudamatpath) {
 
 /* ------------------------------ Utility routines ------------------------------ */
 
-extern int get_leading_dimension(cudamat* mat) {
+EXPORT int get_leading_dimension(cudamat* mat) {
     return mat->is_trans ? mat->size[1] : mat->size[0];
 }
 
-extern int get_nonleading_dimension(cudamat* mat) {
+EXPORT int get_nonleading_dimension(cudamat* mat) {
     return mat->is_trans ? mat->size[0] : mat->size[1];
 }
 
-extern void set_transpose(cudamat* mat, int is_trans) {
+EXPORT void set_transpose(cudamat* mat, int is_trans) {
     mat->is_trans = is_trans;
 }
 
@@ -101,13 +101,13 @@ inline char get_transpose_char(cudamat* mat) {
     return mat->is_trans ? 't' : 'n';
 }
 
-extern void cuda_sync_threads() {
+EXPORT void cuda_sync_threads() {
     cudaThreadSynchronize();
 }
 
 /* ------------------------------ Allocating/moving data ------------------------------ */
 
-extern int allocate_device_memory(cudamat* mat) {
+EXPORT int allocate_device_memory(cudamat* mat) {
     int len = mat->size[0]*mat->size[1];
 
     cublasStatus stat;
@@ -123,7 +123,7 @@ extern int allocate_device_memory(cudamat* mat) {
     return 0;
 }
 
-extern int copy_to_host(cudamat* mat) {
+EXPORT int copy_to_host(cudamat* mat) {
     int len = mat->size[0]*mat->size[1];
 
     if (mat->on_device) {
@@ -137,7 +137,7 @@ extern int copy_to_host(cudamat* mat) {
     return 0;
 }
 
-extern int copy_to_device(cudamat* mat) {
+EXPORT int copy_to_device(cudamat* mat) {
     int len = mat->size[0]*mat->size[1];
     int err_code = 0;
 
@@ -158,7 +158,7 @@ extern int copy_to_device(cudamat* mat) {
     return 0;
 }
 
-extern int copy_on_device(cudamat* mat1, cudamat* mat2) {
+EXPORT int copy_on_device(cudamat* mat1, cudamat* mat2) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (mat1->size[0] != mat2->size[0] || mat1->size[1] != mat2->size[1])
@@ -172,7 +172,7 @@ extern int copy_on_device(cudamat* mat1, cudamat* mat2) {
         return 0;
 }
 
-extern int get_row_slice(cudamat* source, cudamat* target, unsigned int start, unsigned int end) {
+EXPORT int get_row_slice(cudamat* source, cudamat* target, unsigned int start, unsigned int end) {
     int height = source->size[0];
     int width = source->size[1];
 
@@ -193,7 +193,7 @@ extern int get_row_slice(cudamat* source, cudamat* target, unsigned int start, u
         return 0;
 }
 
-extern int set_row_slice(cudamat* source, cudamat* target, unsigned int start, unsigned int end) {
+EXPORT int set_row_slice(cudamat* source, cudamat* target, unsigned int start, unsigned int end) {
     int height = target->size[0];
     int width = target->size[1];
 
@@ -214,7 +214,7 @@ extern int set_row_slice(cudamat* source, cudamat* target, unsigned int start, u
         return 0;
 }
 
-extern int copy_transpose(cudamat* source, cudamat* target) {
+EXPORT int copy_transpose(cudamat* source, cudamat* target) {
     unsigned int height = source->size[0];
     unsigned int width = source->size[1];
 
@@ -241,7 +241,7 @@ extern int copy_transpose(cudamat* source, cudamat* target) {
         return 0;
 }
 
-extern int free_device_memory(cudamat* mat) {
+EXPORT int free_device_memory(cudamat* mat) {
     if (mat->owns_data && mat->on_device) {
         cublasStatus stat;
 
@@ -255,7 +255,7 @@ extern int free_device_memory(cudamat* mat) {
     return 0;
 }
 
-extern int reshape(cudamat* mat, unsigned int m, unsigned int n) {
+EXPORT int reshape(cudamat* mat, unsigned int m, unsigned int n) {
     if (mat->size[0] * mat->size[1] != m * n)
         return ERROR_INCOMPATIBLE_DIMENSIONS;
 
@@ -265,7 +265,7 @@ extern int reshape(cudamat* mat, unsigned int m, unsigned int n) {
     return 0;
 }
 
-extern int get_slice(cudamat* source, cudamat* target, unsigned int first_col, unsigned int last_col) {
+EXPORT int get_slice(cudamat* source, cudamat* target, unsigned int first_col, unsigned int last_col) {
     if (source->is_trans)
         return ERROR_TRANSPOSED;
 
@@ -289,7 +289,7 @@ extern int get_slice(cudamat* source, cudamat* target, unsigned int first_col, u
     return 0;
 }
 
-extern int get_vector_slice(cudamat* source, cudamat* target, unsigned int first_ind, unsigned int last_ind) {
+EXPORT int get_vector_slice(cudamat* source, cudamat* target, unsigned int first_ind, unsigned int last_ind) {
     // source must be a vector
     if (source->size[0] > 1 && source->size[1] > 1)
         return ERROR_GENERIC;
@@ -331,7 +331,7 @@ extern int get_vector_slice(cudamat* source, cudamat* target, unsigned int first
 
 /* ------------------------------ Initialization routines ------------------------------ */
 
-extern void init_from_array(cudamat* mat, float* data, int m, int n) {
+EXPORT void init_from_array(cudamat* mat, float* data, int m, int n) {
     mat->data_host = data;
     mat->size[0] = m;
     mat->size[1] = n;
@@ -341,7 +341,7 @@ extern void init_from_array(cudamat* mat, float* data, int m, int n) {
     mat->owns_data = 1;
 }
 
-extern int init_empty(cudamat* mat, int m, int n) {
+EXPORT int init_empty(cudamat* mat, int m, int n) {
     mat->size[0] = m;
     mat->size[1] = n;
     mat->on_device = 0;
@@ -353,7 +353,7 @@ extern int init_empty(cudamat* mat, int m, int n) {
 }
 
 /* ------------------------------ Random number generation ------------------------------ */
-extern int fill_with_rand(rnd_struct* rnd_state, cudamat* mat) {
+EXPORT int fill_with_rand(rnd_struct* rnd_state, cudamat* mat) {
     int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device)
@@ -370,7 +370,7 @@ extern int fill_with_rand(rnd_struct* rnd_state, cudamat* mat) {
         return 0;
 }
 
-extern int fill_with_randn(rnd_struct* rnd_state, cudamat* mat) {
+EXPORT int fill_with_randn(rnd_struct* rnd_state, cudamat* mat) {
     int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device)
@@ -388,7 +388,7 @@ extern int fill_with_randn(rnd_struct* rnd_state, cudamat* mat) {
 }
 /* ------------------------------ Algebraic operations ------------------------------ */
 
-extern int add_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
+EXPORT int add_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -414,7 +414,7 @@ extern int add_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     return 0;
 }
 
-extern int add_col_mult(cudamat* mat, cudamat* vec, cudamat* target, float mult) {
+EXPORT int add_col_mult(cudamat* mat, cudamat* vec, cudamat* target, float mult) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -439,7 +439,7 @@ extern int add_col_mult(cudamat* mat, cudamat* vec, cudamat* target, float mult)
     return 0;
 }
 
-extern int add_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
+EXPORT int add_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -464,7 +464,7 @@ extern int add_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     return 0;
 }
 
-extern int mult_by_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
+EXPORT int mult_by_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -489,7 +489,7 @@ extern int mult_by_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     return 0;
 }
 
-extern int mult_by_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
+EXPORT int mult_by_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -514,7 +514,7 @@ extern int mult_by_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     return 0;
 }
 
-extern int divide_by_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
+EXPORT int divide_by_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -539,7 +539,7 @@ extern int divide_by_col_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     return 0;
 }
 
-extern int divide_by_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
+EXPORT int divide_by_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -564,7 +564,7 @@ extern int divide_by_row_vec(cudamat* mat, cudamat* vec, cudamat* target) {
     return 0;
 }
 
-extern int less_than(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int less_than(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device)
@@ -588,7 +588,7 @@ extern int less_than(cudamat* mat1, cudamat* mat2, cudamat* target) {
     return 0;
 }
 
-extern int less_than_scalar(cudamat* mat, float val, cudamat* target) {
+EXPORT int less_than_scalar(cudamat* mat, float val, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -611,7 +611,7 @@ extern int less_than_scalar(cudamat* mat, float val, cudamat* target) {
     return 0;
 }
 
-extern int greater_than(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int greater_than(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -635,7 +635,7 @@ extern int greater_than(cudamat* mat1, cudamat* mat2, cudamat* target) {
     return 0;
 }
 
-extern int greater_than_scalar(cudamat* mat, float val, cudamat* target) {
+EXPORT int greater_than_scalar(cudamat* mat, float val, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -658,7 +658,7 @@ extern int greater_than_scalar(cudamat* mat, float val, cudamat* target) {
     return 0;
 }
 
-extern int equals(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int equals(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -682,7 +682,7 @@ extern int equals(cudamat* mat1, cudamat* mat2, cudamat* target) {
     return 0;
 }
 
-extern int equals_scalar(cudamat* mat, float val, cudamat* target) {
+EXPORT int equals_scalar(cudamat* mat, float val, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -705,7 +705,7 @@ extern int equals_scalar(cudamat* mat, float val, cudamat* target) {
     return 0;
 }
 
-extern int minimum(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int minimum(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -728,7 +728,7 @@ extern int minimum(cudamat* mat1, cudamat* mat2, cudamat* target) {
     return 0;
 }
 
-extern int minimum_scalar(cudamat* mat, float val, cudamat* target) {
+EXPORT int minimum_scalar(cudamat* mat, float val, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -750,7 +750,7 @@ extern int minimum_scalar(cudamat* mat, float val, cudamat* target) {
     return 0;
 }
 
-extern int maximum(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int maximum(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -773,7 +773,7 @@ extern int maximum(cudamat* mat1, cudamat* mat2, cudamat* target) {
     return 0;
 }
 
-extern int maximum_scalar(cudamat* mat, float val, cudamat* target) {
+EXPORT int maximum_scalar(cudamat* mat, float val, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -795,7 +795,7 @@ extern int maximum_scalar(cudamat* mat, float val, cudamat* target) {
     return 0;
 }
 
-extern int min_by_axis(cudamat* mat, cudamat* target, int axis) {
+EXPORT int min_by_axis(cudamat* mat, cudamat* target, int axis) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -826,7 +826,7 @@ extern int min_by_axis(cudamat* mat, cudamat* target, int axis) {
     return 0;
 }
 
-extern int max_by_axis(cudamat* mat, cudamat* target, int axis) {
+EXPORT int max_by_axis(cudamat* mat, cudamat* target, int axis) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -857,7 +857,7 @@ extern int max_by_axis(cudamat* mat, cudamat* target, int axis) {
     return 0;
 }
 
-extern int argmin_by_axis(cudamat* mat, cudamat* target, int axis) {
+EXPORT int argmin_by_axis(cudamat* mat, cudamat* target, int axis) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -888,7 +888,7 @@ extern int argmin_by_axis(cudamat* mat, cudamat* target, int axis) {
     return 0;
 }
 
-extern int argmax_by_axis(cudamat* mat, cudamat* target, int axis) {
+EXPORT int argmax_by_axis(cudamat* mat, cudamat* target, int axis) {
     unsigned int h = mat->size[0],
                  w = mat->size[1];
 
@@ -919,7 +919,7 @@ extern int argmax_by_axis(cudamat* mat, cudamat* target, int axis) {
     return 0;
 }
 
-extern int sign(cudamat* mat, cudamat* target) {
+EXPORT int sign(cudamat* mat, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -942,7 +942,7 @@ extern int sign(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_sigmoid(cudamat* mat, cudamat* target) {
+EXPORT int apply_sigmoid(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -962,7 +962,7 @@ extern int apply_sigmoid(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_tanh(cudamat* mat, cudamat* target) {
+EXPORT int apply_tanh(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -982,7 +982,7 @@ extern int apply_tanh(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_soft_threshold(cudamat* mat, float alpha, cudamat* target) {
+EXPORT int apply_soft_threshold(cudamat* mat, float alpha, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1001,7 +1001,7 @@ extern int apply_soft_threshold(cudamat* mat, float alpha, cudamat* target) {
     return 0;
 }
 
-extern int apply_abs(cudamat* mat, cudamat* target) {
+EXPORT int apply_abs(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1021,7 +1021,7 @@ extern int apply_abs(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_log_1_plus_exp(cudamat* mat, cudamat* target) {
+EXPORT int apply_log_1_plus_exp(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1041,7 +1041,7 @@ extern int apply_log_1_plus_exp(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_log(cudamat* mat, cudamat* target) {
+EXPORT int apply_log(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1061,7 +1061,7 @@ extern int apply_log(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_exp(cudamat* mat, cudamat* target) {
+EXPORT int apply_exp(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1081,7 +1081,7 @@ extern int apply_exp(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_gamma(cudamat* mat, cudamat* target) {
+EXPORT int apply_gamma(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1101,7 +1101,7 @@ extern int apply_gamma(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_lgamma(cudamat* mat, cudamat* target) {
+EXPORT int apply_lgamma(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1121,7 +1121,7 @@ extern int apply_lgamma(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_sqrt(cudamat* mat, cudamat* target) {
+EXPORT int apply_sqrt(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1141,7 +1141,7 @@ extern int apply_sqrt(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int apply_pow(cudamat* mat, float pow, cudamat* target) {
+EXPORT int apply_pow(cudamat* mat, float pow, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1161,7 +1161,7 @@ extern int apply_pow(cudamat* mat, float pow, cudamat* target) {
     return 0;
 }
 
-extern int apply_pow_matrix(cudamat* mat, cudamat* pow, cudamat* target) {
+EXPORT int apply_pow_matrix(cudamat* mat, cudamat* pow, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1184,7 +1184,7 @@ extern int apply_pow_matrix(cudamat* mat, cudamat* pow, cudamat* target) {
     return 0;
 }
 
-extern int reciprocal(cudamat* mat, cudamat* target) {
+EXPORT int reciprocal(cudamat* mat, cudamat* target) {
     unsigned int len = mat->size[0] * mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1204,7 +1204,7 @@ extern int reciprocal(cudamat* mat, cudamat* target) {
     return 0;
 }
 
-extern int dot(cudamat* mat1, cudamat* mat2, cudamat* target, float beta, float alpha) {
+EXPORT int dot(cudamat* mat1, cudamat* mat2, cudamat* target, float beta, float alpha) {
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
         return ERROR_NOT_ON_DEVICE;
 
@@ -1232,7 +1232,7 @@ extern int dot(cudamat* mat1, cudamat* mat2, cudamat* target, float beta, float 
     return 0;
 }
 
-extern float vdot(cudamat* mat1, cudamat* mat2, int* err_code) {
+EXPORT float vdot(cudamat* mat1, cudamat* mat2, int* err_code) {
     int len = mat1->size[0]*mat1->size[1];
     float res;
 
@@ -1262,7 +1262,7 @@ extern float vdot(cudamat* mat1, cudamat* mat2, int* err_code) {
 
 /* Perform the operation mat1 = mat1 + alpha * mat2. mat1 and mat2 must
    have the same transposedness. */
-extern int add_mult(cudamat* mat1, cudamat* mat2, float alpha) {
+EXPORT int add_mult(cudamat* mat1, cudamat* mat2, float alpha) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device)
@@ -1282,7 +1282,7 @@ extern int add_mult(cudamat* mat1, cudamat* mat2, float alpha) {
     return 0;
 }
 
-extern int add_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int add_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -1314,7 +1314,7 @@ extern int add_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
      return 0;
 }
 
-extern int subtract_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int subtract_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -1338,7 +1338,7 @@ extern int subtract_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
     return 0;
 }
 
-extern int divide_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int divide_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -1363,7 +1363,7 @@ extern int divide_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
 }
 
 /* Elementwise multiplication of 2 matrices */
-extern int mult_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
+EXPORT int mult_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
     int len = mat1->size[0]*mat1->size[1];
 
     if (!mat1->on_device || !mat2->on_device || !target->on_device)
@@ -1387,7 +1387,7 @@ extern int mult_elementwise(cudamat* mat1, cudamat* mat2, cudamat* target) {
     return 0;
 }
 
-extern int assign_scalar(cudamat* mat, float alpha) {
+EXPORT int assign_scalar(cudamat* mat, float alpha) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device)
@@ -1404,7 +1404,7 @@ extern int assign_scalar(cudamat* mat, float alpha) {
     return 0;
 }
 
-extern int mult_by_scalar(cudamat* mat, float alpha, cudamat* target) {
+EXPORT int mult_by_scalar(cudamat* mat, float alpha, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1432,7 +1432,7 @@ extern int mult_by_scalar(cudamat* mat, float alpha, cudamat* target) {
     return 0;
 }
 
-extern int divide_by_scalar(cudamat* mat, float alpha, cudamat* target) {
+EXPORT int divide_by_scalar(cudamat* mat, float alpha, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1452,7 +1452,7 @@ extern int divide_by_scalar(cudamat* mat, float alpha, cudamat* target) {
     return 0;
 }
 
-extern int add_scalar(cudamat* mat, float alpha, cudamat* target) {
+EXPORT int add_scalar(cudamat* mat, float alpha, cudamat* target) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device || !target->on_device)
@@ -1472,7 +1472,7 @@ extern int add_scalar(cudamat* mat, float alpha, cudamat* target) {
     return 0;
 }
 
-extern float euclid_norm(cudamat* mat, int* err_code) {
+EXPORT float euclid_norm(cudamat* mat, int* err_code) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device) {
@@ -1491,7 +1491,7 @@ extern float euclid_norm(cudamat* mat, int* err_code) {
     }
 }
 
-extern float manhattan_norm(cudamat* mat, int* err_code) {
+EXPORT float manhattan_norm(cudamat* mat, int* err_code) {
     int len = mat->size[0]*mat->size[1];
 
     if (!mat->on_device) {
@@ -1510,7 +1510,7 @@ extern float manhattan_norm(cudamat* mat, int* err_code) {
     }
 }
 
-extern int selectRows(cudamat* source, cudamat* target, cudamat* indices){
+EXPORT int selectRows(cudamat* source, cudamat* target, cudamat* indices){
     const int nRetRows = indices->size[1];
 
     if (nRetRows==0) return 0;
@@ -1529,7 +1529,7 @@ extern int selectRows(cudamat* source, cudamat* target, cudamat* indices){
         return 0;
 }
 
-extern int setSelectedRows(cudamat* target, cudamat* source, cudamat* indices){
+EXPORT int setSelectedRows(cudamat* target, cudamat* source, cudamat* indices){
     const int nSetRows = indices->size[1];
 
     if (nSetRows==0)
@@ -1549,7 +1549,7 @@ extern int setSelectedRows(cudamat* target, cudamat* source, cudamat* indices){
         return 0;
 }
 
-extern int where(cudamat* condition_mat, cudamat* if_mat, cudamat* else_mat, cudamat* target) {
+EXPORT int where(cudamat* condition_mat, cudamat* if_mat, cudamat* else_mat, cudamat* target) {
     unsigned int len = condition_mat->size[0] * condition_mat->size[1];
 
     if (!condition_mat->on_device || !target->on_device)

--- a/cudamat/cudamat.cuh
+++ b/cudamat/cudamat.cuh
@@ -1,3 +1,9 @@
+#if defined(_WIN32) || defined(__CYGWIN__)
+  #define EXPORT __declspec(dllexport)
+#else
+  #define EXPORT __attribute__ ((visibility("default")))
+#endif
+
 #define SYNC_THREADS 1
 
 #define ERROR_INCOMPATIBLE_DIMENSIONS -1

--- a/cudamat/cudamat.py
+++ b/cudamat/cudamat.py
@@ -6,12 +6,16 @@ import sysconfig
 import ctypes as ct
 import numpy as np
 
+def load_library(basename):
+    if platform.system() == 'Windows':
+       ext = '.dll'
+    else:
+       ext = sysconfig.get_config_var('SO')
+    return ct.cdll.LoadLibrary(os.path.join(
+        os.path.dirname(__file__) or os.path.curdir,
+        basename + ext))
 
-if platform.system() == 'Windows':
-    _cudamat = ct.cdll.LoadLibrary('libcudamat.dll')
-else:
-    _cudamat = ct.cdll.LoadLibrary(os.path.join(
-        os.path.dirname(__file__) or os.path.curdir, 'libcudamat' + sysconfig.get_config_var('SO')))
+_cudamat = load_library('libcudamat')
 
 _cudamat.get_last_cuda_error.restype = ct.c_char_p
 _cudamat.cublas_init.restype = ct.c_int

--- a/cudamat/learn.cu
+++ b/cudamat/learn.cu
@@ -14,7 +14,7 @@ inline bool checkCUDAError() {
     return cudaSuccess != err;
 }
 
-extern int mult_by_sigmoid_deriv(cudamat* target, cudamat* acts) {
+EXPORT int mult_by_sigmoid_deriv(cudamat* target, cudamat* acts) {
     int len = acts->size[0]*acts->size[1];
 
     if (acts->is_trans != target->is_trans)

--- a/cudamat/learn.py
+++ b/cudamat/learn.py
@@ -1,17 +1,11 @@
 import os
-import pdb
-import platform
-import warnings
-import sysconfig
+
 import ctypes as ct
 import numpy as np
-from cudamat import generate_exception
 
-if platform.system() == 'Windows':
-    _cudalearn = ct.cdll.LoadLibrary('libcudalearn.dll')
-else:
-    _cudalearn = ct.cdll.LoadLibrary(os.path.join(os.path.dirname(__file__) or os.path.curdir,
-                                                  'libcudalearn' + sysconfig.get_config_var('SO')))
+from cudamat import load_library, generate_exception
+
+_cudalearn = load_library('libcudalearn')
 
 _cudalearn.mult_by_sigmoid_deriv.restype = ct.c_int
 

--- a/setup.py
+++ b/setup.py
@@ -1,15 +1,16 @@
 #!/usr/bin/env python
 
 import os
-from distutils.spawn import spawn
+# on Windows, we need the original PATH without Anaconda's compiler in it:
+PATH = os.environ.get('PATH')
+from distutils.spawn import spawn, find_executable
 from setuptools import setup, find_packages, Extension
 from setuptools.command.build_ext import build_ext
 import sys
 
 # CUDA specific config
 # nvcc is assumed to be in user's PATH
-nvcc_compile_args = ['-O', '--ptxas-options=-v', '--compiler-options',
-                     "'-fPIC'"]
+nvcc_compile_args = ['-O', '--ptxas-options=-v', '--compiler-options=-fPIC']
 nvcc_compile_args = os.environ.get('NVCCFLAGS', '').split() + nvcc_compile_args
 cuda_libs = ['cublas']
 
@@ -34,6 +35,8 @@ class CUDA_build_ext(build_ext):
         self.compiler.src_extensions.append('.cu')
         self.compiler.set_executable('compiler_so', 'nvcc')
         self.compiler.set_executable('linker_so', 'nvcc --shared')
+        if hasattr(self.compiler, '_c_extensions'):
+            self.compiler._c_extensions.append('.cu')  # needed for Windows
         self.compiler.spawn = self.spawn
         build_ext.build_extensions(self)
 
@@ -53,6 +56,57 @@ class CUDA_build_ext(build_ext):
                     del cmd[index:index+2]
                 except ValueError:
                     break
+        elif self.compiler.compiler_type == 'msvc':
+            # There are several things we need to do to change the commands
+            # issued by MSVCCompiler into one that works with nvcc. In the end,
+            # it might have been easier to write our own CCompiler class for
+            # nvcc, as we're only interested in creating a shared library to
+            # load with ctypes, not in creating an importable Python extension.
+            # - First, we replace the cl.exe or link.exe call with an nvcc
+            #   call. In case we're running Anaconda, we search cl.exe in the
+            #   original search path we captured further above -- Anaconda
+            #   inserts a MSVC version into PATH that is too old for nvcc.
+            cmd[:1] = ['nvcc', '--compiler-bindir',
+                       os.path.dirname(find_executable("cl.exe", PATH))
+                       or cmd[0]]
+            # - Secondly, we fix a bunch of command line arguments.
+            for idx, c in enumerate(cmd):
+                # create .dll instead of .pyd files
+                if '.pyd' in c: cmd[idx] = c = c.replace('.pyd', '.dll')
+                # replace /c by -c
+                if c == '/c': cmd[idx] = '-c'
+                # replace /DLL by --shared
+                elif c == '/DLL': cmd[idx] = '--shared'
+                # remove --compiler-options=-fPIC
+                elif '-fPIC' in c: del cmd[idx]
+                # replace /Tc... by ...
+                elif c.startswith('/Tc'): cmd[idx] = c[3:]
+                # replace /Fo... by -o ...
+                elif c.startswith('/Fo'): cmd[idx:idx+1] = ['-o', c[3:]]
+                # replace /LIBPATH:... by -L...
+                elif c.startswith('/LIBPATH:'): cmd[idx] = '-L' + c[9:]
+                # replace /OUT:... by -o ...
+                elif c.startswith('/OUT:'): cmd[idx:idx+1] = ['-o', c[5:]]
+                # remove /EXPORT:initlibcudamat or /EXPORT:initlibcudalearn
+                elif c.startswith('/EXPORT:'): del cmd[idx]
+                # replace cublas.lib by -lcublas
+                elif c == 'cublas.lib': cmd[idx] = '-lcublas'
+            # - Finally, we pass on all arguments starting with a '/' to the
+            #   compiler or linker, and have nvcc handle all other arguments
+            if '--shared' in cmd:
+                pass_on = '--linker-options='
+                # we only need MSVCRT for a .dll, remove CMT if it sneaks in:
+                cmd.append('/NODEFAULTLIB:libcmt.lib')
+            else:
+                pass_on = '--compiler-options='
+            cmd = ([c for c in cmd if c[0] != '/'] +
+                   [pass_on + ','.join(c for c in cmd if c[0] == '/')])
+            # For the future: Apart from the wrongly set PATH by Anaconda, it
+            # would suffice to run the following for compilation on Windows:
+            # nvcc -c -O -o <file>.obj <file>.cu
+            # And the following for linking:
+            # nvcc --shared -o <file>.dll <file1>.obj <file2>.obj -lcublas
+            # This could be done by a NVCCCompiler class for all platforms.
         spawn(cmd, search_path, verbose, dry_run)
 
 setup(name="cudamat",

